### DISCUSSION TITLE
Foundry Fund Me Tests Failing with zkSync Era Mainnet PriceFeed/RPC

### DISCUSSION THREAD

**Discussion by mackcee** [Cyfrin/foundry-full-course-cu#1473] [Q&A]
Hey Everyone. I'm really enjoying the course so far.

I have run into an issue where my tests are failing when I am trying to use the `--fork-url https://rpc.ankr.com/zksync_era`

I know that RPC url is working because it returns the correct chain id and therefore priceFeed contract address from my helper config:

```solidity
    constructor() {
        if (block.chainid == 11155111) {
            activeNetworkConfig = getSepoliaEthConfig();
        } else if (block.chainid == 1) {
            activeNetworkConfig = getMainnetEthConfig();
        } else if (block.chainid == 324) {
            activeNetworkConfig = getzkSyncEthConfig();
        } else {
            activeNetworkConfig = getAnvilEthConfig();
        }
    }
    
        function getzkSyncEthConfig() public pure returns (NetworkConfig memory) {
        return
            NetworkConfig({
                priceFeed: 0x6D41d1dc818112880b40e26BD6FD347E41008eDA
            });
    }

 ```

When I call my tests I get this when trying to assert the version, which implies that the pricefeed contract is wrong:

```
  [10135] FundMeTest::testPriceFeedVersionIsAccurate()
    ├─ [5099] FundMe::getVersion() [staticcall]
    │   ├─ [0] 0x6D41d1dc818112880b40e26BD6FD347E41008eDA::version() [staticcall]
    │   │   └─ ← ()
    │   └─ ← EvmError: Revert
    └─ ← EvmError: Revert
```

However I can curl that pricefeed contract with the RPC URL and the ABI-encoded data just fine:

```
 ~/Pr/B/F/foundry-fund-me │ main +2 !4 ?5  curl -X POST https://rpc.ankr.com/zksync_era -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"eth_call","params":[{"to":"0x6D41d1dc818112880b40e26BD6FD347E41008eDA","data":"0x54fd4d50"},"latest"],"id":1}'

{"jsonrpc":"2.0","result":"0x0000000000000000000000000000000000000000000000000000000000000004","id":1}% 
```

Can anyone suggest why it might not be working?

Thanks!

**Comment by EngrPips** [comment]
Hello @Charlie-Mack , it is difficult for me to see what's going wrong on here as everything seems right, but i always get `EvmError: Revert` when i call a wrong method on an interface or i wrap a contract address with a wrong interface. But then it doesn't seem like that is the situation here.

**Reply by mackcee** [reply]
Yeah the code doesnt change between chains, I'm using the same method: `version()` with each chain so it can't be that.

The contract address looks right and worked with curl and the same with the RPC URL I am using. Its very strange.

**Reply by EngrPips** [reply]
Yeah, that is very strange. but what i would always do in this kind of situation is to import `console2` into the contract i am testing and try console log at different point of the function i am testing to see where exactly in that function did the `EVM` revert.
